frcxx_nft/src/nft/state.rs

```diff
@@ -0,0 +1,98 @@
+use cid::multihash::Code;
+use cid::Cid;
+use fvm_ipld_amt::Amt;
+use fvm_ipld_amt::Error as AmtError;
+use fvm_ipld_blockstore::Block;
+use fvm_ipld_blockstore::Blockstore;
+use fvm_ipld_encoding::tuple::*;
+use fvm_ipld_encoding::CborStore;
+use fvm_ipld_encoding::DAG_CBOR;
+use fvm_shared::ActorID;
+use thiserror::Error;
+
+pub type TokenID = u64;
+
+#[derive(Serialize_tuple, Deserialize_tuple, Debug)]
+pub struct BatchMintReturn {
+    pub tokens: Vec<TokenID>,
+}
+
+#[derive(Error, Debug)]
+pub enum StateError {
+    #[error("ipld hamt error: {0}")]
+    IpldAmt(#[from] AmtError),
+    #[error("other error: {0}")]
+    Other(String),
+}
+
+/// NFT state IPLD structure
+#[derive(Serialize_tuple, Deserialize_tuple, PartialEq, Eq, Clone, Debug)]
+pub struct NFTSetState {
+    /// Amt<ActorID> of balances as a Hamt where index is TokenID
+    pub tokens: Cid,
```

Comment:
Maybe `owners`? `owners_by_token`? `token_owners`? Others might also have an owner->tokens map, which one might call `tokens`. Tho both ways can be confused.